client/sites-dashboard/use-sites-data-query.ts

```diff
@@ -0,0 +1,45 @@
+import config from '@automattic/calypso-config';
+import { useQuery } from 'react-query';
+import wpcom from 'calypso/lib/wp';
+import { SiteData, SiteDataOptions } from 'calypso/state/ui/selectors/site-data';
+
+// Performance-optimized request for lists of sites.
+// Don't add more fields because you will make the request slower.
+export const SITE_EXCERPT_REQUEST_FIELDS = [
+	'ID',
+	'URL',
+	'is_coming_soon',
+	'is_private',
+	'launch_status',
+	'slug',
+	'icon',
+	'name',
+	'options',
+	'plan',
+] as const;
+
+export const SITE_EXCERPT_REQUEST_OPTIONS = [ 'is_wpforteams_site' ] as const;
+
+export type SiteExcerptData = Pick< SiteData, typeof SITE_EXCERPT_REQUEST_FIELDS[ number ] > & {
+	options: Pick< SiteDataOptions, typeof SITE_EXCERPT_REQUEST_OPTIONS[ number ] >;
+};
+
+const fetchSites = (): Promise< { sites: SiteExcerptData[] } > => {
+	const siteFilter = config< string[] >( 'site_filter' );
+	return wpcom.me().sites( {
+		apiVersion: '1.2',
+		site_visibility: 'all',
+		include_domain_only: true,
+		site_activity: 'active',
+		fields: SITE_EXCERPT_REQUEST_FIELDS.join( ',' ),
+		options: SITE_EXCERPT_REQUEST_OPTIONS.join( ',' ),
+		filters: siteFilter.length > 0 ? siteFilter.join( ',' ) : undefined,
+	} );
+};
+
+export const useSitesDataQuery = () =>
+	useQuery( [ 'sites-dashboard-sites-data' ], fetchSites, {
```

Comment:
Hey Daniel, for #1 that's what I came up with:

```diff
diff --git a/client/sites-dashboard/components/sites-dashboard.tsx b/client/sites-dashboard/components/sites-dashboard.tsx
index 649b2c9daf..45e192aa46 100644
--- a/client/sites-dashboard/components/sites-dashboard.tsx
+++ b/client/sites-dashboard/components/sites-dashboard.tsx
@@ -59,9 +59,9 @@ const DashboardHeading = styled.h1`
 
 export function SitesDashboard( { launchStatus }: SitesDashboardProps ) {
 	const { __ } = useI18n();
-	const { data: sites } = useSitesDataQuery();
+	const { data: sites, isLoading } = useSitesDataQuery();
 
-	if ( ! sites ) {
+	if ( ! sites || isLoading ) {
 		return null;
 	}
 
diff --git a/client/sites-dashboard/use-sites-data-query.ts b/client/sites-dashboard/use-sites-data-query.ts
index 50ed22c7db..256842faa2 100644
--- a/client/sites-dashboard/use-sites-data-query.ts
+++ b/client/sites-dashboard/use-sites-data-query.ts
@@ -1,7 +1,10 @@
 import config from '@automattic/calypso-config';
 import { useQuery } from 'react-query';
+import { useSelector } from 'react-redux';
 import wpcom from 'calypso/lib/wp';
+import getSites from 'calypso/state/selectors/get-sites';
 import { SiteData, SiteDataOptions } from 'calypso/state/ui/selectors/site-data';
+import { notNullish } from './util';
 
 // Performance-optimized request for lists of sites.
 // Don't add more fields because you will make the request slower.
@@ -21,7 +24,7 @@ export const SITE_EXCERPT_REQUEST_FIELDS = [
 export const SITE_EXCERPT_REQUEST_OPTIONS = [ 'is_wpforteams_site' ] as const;
 
 export type SiteExcerptData = Pick< SiteData, typeof SITE_EXCERPT_REQUEST_FIELDS[ number ] > & {
-	options: Pick< SiteDataOptions, typeof SITE_EXCERPT_REQUEST_OPTIONS[ number ] >;
+	options?: Pick< SiteDataOptions, typeof SITE_EXCERPT_REQUEST_OPTIONS[ number ] >;
 };
 
 const fetchSites = (): Promise< { sites: SiteExcerptData[] } > => {
@@ -37,9 +40,18 @@ const fetchSites = (): Promise< { sites: SiteExcerptData[] } > => {
 	} );
 };
 
-export const useSitesDataQuery = () =>
-	useQuery( [ 'sites-dashboard-sites-data' ], fetchSites, {
-		enabled: true,
+export const useSitesDataQuery = () => {
+	const sites = useSelector( getSites );
+
+	const reduxData = {
+		sites: sites?.filter( notNullish ),
+	};
+
+	const result = useQuery( [ 'sites-dashboard-sites-data' ], fetchSites, {
 		staleTime: 1000 * 60 * 5, // 5 minutes
-		select: ( data ) => data?.sites ?? [],
+		select: ( data ) => data?.sites,
+		placeholderData: reduxData,
 	} );
+
+	return result;
+};
```

It should be pretty straightforward, but let me know if you run into questions.

For #2: what exactly do you want to achieve?

Reply:
> It should be pretty straightforward, but let me know if you run into questions.

Thanks for the example, @zaguiini !

Incorporating some learnings in our Slack chat, here's what I ended up with: 61a4fa18535a52561d86aebe992b185651ca14e3

Notably, I'm using `useStore().getState().sites.items` because I only want to access the _initial_ state of the store. When I use `useSelector( getSites )`, `useSitesDataQuery` is called every time the store updates:

![image](https://user-images.githubusercontent.com/36432/179603260-eb4c974d-6f35-4ead-a8da-b0bf1ce4f0d2.png)

> For 2: what exactly do you want to achieve?

When `useStore().getState().sites.items` is updated, I want to make sure the `react-query` cache is cleared.

Reply:
😕 I'm a little confused because it seems contradictory. We're using `useStore().getState().sites.items` because we only want the initial state. But then you ask how do I know when it's updated, so it sounds like you _do_ want to do something every time it changes...

It seems to be working fine without clearing the `react-query` cache, so I think I'll merge with this part of it working as is.